compiler/src/yul/mappers/expressions.rs

```diff
@@ -62,6 +62,27 @@ pub fn expr(
     }
 }
 
+
+pub fn expr_bin_operation<'a>(exp: &Spanned<vyp::Expr<'a>>) -> Result<ExtExpression, CompileError> {
+    if let vyp::Expr::BinOperation{ left, op, right } = &exp.node {
+        // FIXME: How do I get what I need for the `add`
+        let yul_left = literal_expression! {(left.node)};
+        let yul_right = literal_expression! {(right.node)};
```

Comment:
looks good so far!

The correct left and right expressions can be found by calling `expr(...)` again, like we do [here](https://github.com/ethereum/fe/blob/4bd735adac4df4f7cf805a07de7a2348aee6ec0a/compiler/src/yul/mappers/expressions.rs#L154).

`let yul_left = expr(Rc::clone(&scope), left)?;`

We just need to pass in the scope `expr_bin_operation(scope: Shared<FunctionScope>, ...)`.

FYI: `Shared<T>` is an alias for `Rc<RefCell<T>>`. The pattern is described in more detail [here](https://doc.rust-lang.org/book/ch15-05-interior-mutability.html#having-multiple-owners-of-mutable-data-by-combining-rct-and-refcellt). You won't need to do anything with scope here other than providing it to the `expr` call.

Let me know if what I've suggested doesn't work.

Reply:
Thanks, it worked!

>FYI: Shared<T> is an alias for Rc<RefCell<T>>. The pattern is described in more detail here. 

Thanks, I'm loosely familiar with the pattern. :+1:

Reply:
awesome!